app/Http/Controllers/ThreadsController.php

```diff
@@ -63,7 +64,12 @@ public function store(Recaptcha $recaptcha)
         request()->validate([
             'title' => 'required|spamfree',
             'body' => 'required|spamfree',
-            'channel_id' => 'required|exists:channels,id',
+            'channel_id' => [
+                'required',
+                Rule::exists('channels', 'id')->where(function ($query) {
+                    $query->where('archived', false);
+                }),
```

Comment:
Personal preference but since all the other rules are defined here in string format the exists rule could be written as: `'channel_id' => 'required|exists:channels,id,active,true'`